### PR TITLE
Add option to skip dot generation

### PR DESCRIPTION
**Pull request by changusmc**
In a large project with 350 nodes and 8000+ edges, it takes 4 hours to convert the `dot` file to `svg`. This adds an option to leave the `dot` file without an post processing which takes a few seconds.

### CODE REVIEW

**Review by savvasdalkitsis — CHANGES_REQUESTED** [review]
Looks good to me but i think the name `autoGenerateDot` might be a bit confusing as it might indicate that the dot file itself won't be generated.

Can i recommend something like `transformDot`?

File: src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy [hunk 1]
```diff
@@ -76,14 +76,20 @@ class ModuleDependencyGraphPlugin implements Plugin<Project> {
                     autoOpenGraph = Boolean.parseBoolean(project.property('autoOpenGraph').toString())
                 }
                 dotFile.write(dot)
-                project.exec {
-                    executable = "dot"
-                    args("-o", graphOutputFile.absolutePath, "-T$graphOutputFormat", dotFile.absolutePath)
+                def autoGenerateDot = true
```
Comment (changusmc) [inline src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy]:
```suggestion
                def transformDot = true
```

File: src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy [hunk 2]
```diff
@@ -76,14 +76,20 @@ class ModuleDependencyGraphPlugin implements Plugin<Project> {
...
+                if (project.hasProperty('autoGenerateDot')) {
```
Comment (changusmc) [inline src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy]:
```suggestion
                if (project.hasProperty('transformDot')) {
```

File: src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy [hunk 3]
```diff
@@ -76,14 +76,20 @@ class ModuleDependencyGraphPlugin implements Plugin<Project> {
...
+                    autoGenerateDot = Boolean.parseBoolean(project.property('autoGenerateDot').toString())
```
Comment (changusmc) [inline src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy]:
```suggestion
                    transformDot = Boolean.parseBoolean(project.property('transformDot').toString())
```

File: src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy [hunk 4]
```diff
@@ -76,14 +76,20 @@ class ModuleDependencyGraphPlugin implements Plugin<Project> {
...
                 }
-                println("Generaged dot file at: ${dotFile.absolutePath}")
-                println("Generaged output file at: ${graphOutputFile.absolutePath}")
+                if (autoGenerateDot) {
```
Comment (changusmc) [inline src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy]:
```suggestion
                if (transformDot) {
```

File: src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy [hunk 5]
```diff
@@ -76,14 +76,20 @@ class ModuleDependencyGraphPlugin implements Plugin<Project> {
...
+                    project.exec {
+                        executable = "dot"
+                        args("-o", graphOutputFile.absolutePath, "-T$graphOutputFormat", dotFile.absolutePath)
+                    }
+                    println("Generated output file at: ${graphOutputFile.absolutePath}")
+                }
+                println("Generated dot file at: ${dotFile.absolutePath}")
 
-                if (autoOpenGraph) {
+                if (autoGenerateDot && autoOpenGraph) {
```
Comment (changusmc) [inline src/main/groovy/com/savvasdalkitsis/module/deps/graph/ModuleDependencyGraphPlugin.groovy]:
```suggestion
                if (transformDot && autoOpenGraph) {
```

File: README.md [hunk 6]
```diff
@@ -96,6 +96,22 @@ or in the `gradle.properties` file:
 autoOpenGraph=false
 ```
 
+### Auto generate dot file
```
Comment (changusmc) [inline README.md]:
```suggestion
### Transform dot file
```

File: README.md [hunk 7]
```diff
@@ -96,6 +96,22 @@ or in the `gradle.properties` file:
...
+
+By default, the command line tool `dot` will be used to generate an output defined in the
```
Comment (changusmc) [inline README.md]:
```suggestion
By default, the command line tool `dot` will be used to transform an output defined in the
```

File: README.md [hunk 8]
```diff
@@ -96,6 +96,22 @@ or in the `gradle.properties` file:
...
+Gradle property `graphOutputFormat`. If you don't want this to happen, 
+you can specify the following parameter:
+
+```bash
+./gradlew graphModules -PautoGenerateDot=false
```
Comment (changusmc) [inline README.md]:
```suggestion
./gradlew graphModules -PtransformDot=false      
```

File: README.md [hunk 9]
```diff
@@ -96,6 +96,22 @@ or in the `gradle.properties` file:
...
+you can specify the following parameter:
+
+```bash
+./gradlew graphModules -PautoGenerateDot=false      
+```
+
+or in the `gradle.properties` file:
+
+```
+autoGenerateDot=false
```
Comment (changusmc) [inline README.md]:
```suggestion
transformDot=false
```

**Comment by savvasdalkitsis** [comment]
Published in v0.12

**Pull request merged**